### DISCUSSION TITLE
Indentation (tabs vs. spaces) with `lsp-format-buffer` and `lua-mode`

### DISCUSSION THREAD

**Discussion by AndreasMatthias** [emacs-lsp/lsp-mode#3980] [Q&A] [answered]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [X] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

`lua-mode` in default configuration uses spaces for indentation, but `(lsp-format-buffer)` uses tabs.

### Steps to reproduce

Take this example:

```lua
function foo()
print('foo')
end
```

With point at line 2, `(indent-for-tab-command)` indents this line by 3 spaces, because `lua-indent-level` is set to 3. This verifies that **lua-mode** is using spaces and not tabs (in the default configuration).

But if I let **lua-language-server** format the buffer with `(lsp-format-buffer)`, the second line gets indented by one tab and not by 3 spaces.

If I understand correctly **lua-language-server** does not decide by itself whether to use spaces or tabs for indentation, but gets this information from the client. 

### Expected behavior

Indentation of `lua-mode` and `(lsp-format-buffer)` should be the same.

### Which Language Server did you use?

lua-language-server

### OS

Linux

**Comment by yyoncho** [comment, marked as answer]
We use lua-indent-level to get the tab size. (see https://github.com/yyoncho/lsp-mode/blob/inlay-hints/lsp-mode.el#L5910 )
We use indent-tabs-mode to determine whether to ask the server to indent with tabs or not.

**Reply by AndreasMatthias** [reply]
Ok, I see.  I get the expected result if I do
```
(setq-default indent-tabs-mode nil)
```

Though I do not need this setting with `lua-mode`, I must use it in combination with `lsp-mode`. Good to know.
